### DISCUSSION TITLE
nunit3-netcore-console.dll is not finding tests in all my assemblies

### DISCUSSION THREAD

**Discussion by kurtsmack** [nunit/nunit-console#1357] [Q&A] [answered]
I'm using nunit3-netcore-console 3.16.3 with --explore to list the tests in each of my C# test assemblies. I have about 96 assemblies. It works great except that 7 of them return no tests.

I compared the csproj for one that doesn't work to a very similar one that does and I can't see any real difference. They both target net6, both use "Microsoft.NET.Test.Sdk" Version="17.7.0", "NUnit" Version="3.13.3" and "NUnit3TestAdapter" Version="4.5.0". There's no redirection in App.config. The command looks like this:

`dotnet nunit3-netcore-console.dll --noheader --explore Tests.dll`

I have the same results on both Windows and Linux. I can't use `dotnet test --list-tests` because I want to use filtering too. When I run the tests, I get errors that I'm not sure how to resolve and not sure if they are the reason --explore won't work since I get errors like this trying to run an assembly that does list tests. I don't necessarily care about the errors if --explore works since I won't be running the tests this way. This is one example:

```
Unable to load one or more of the requested types.
Could not load file or assembly 'System.Data.Common, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (0x80131058)
```

The same problem happens with nunit3-console.exe 3.10.0. When I run the tests using `dotnet test`, they run fine. I can also list the lists fine using `dotnet test --list-tests`.

**Comment by CharliePoole** [comment, marked as answer]
A few things you might try...

1. Eliminate references to both `Microsoft.NET.Test.Sdk` and `NUnit3TestAdapter`. They are not used by `nunit-netcore-console`. In particular, the adapter may be pulling in a different version of the NUnit engine.
2. Keep the references, but use a compatible version of the console runner... i.e. the same version as that of the `nunit.engine` included with the adapter. (Sorry, I don't know what that is for 4.5)
3. Use `nunit3-console.exe` instead of `nunit3-netcore-console.dll`. However, use a later version, which supports .NET 6.0. I suggest one of the 3.15 or 3.16 versions.

One question... are any of your failing assemblies using one of these?
* Windows Forms
* WPF
* ASP.NET Core

**Reply by kurtsmack** [reply]
Thanks for the reply. I don't want to use option 3 since I want to run on Linux and not use Mono. I'll try the other options on Monday. The assemblies are not using any of those three things.

**Reply by kurtsmack** [reply]
Thanks again Charlie! I got it working. Option 2 was the key. Instead of using nunit3-netcore-console.dll, I used nunit3-console.dll in the net6.0 folder for NUnit3.Console 3.15.4, which is the version of the engine for NUnit3TestAdapter 4.5.0. Previously, I was using NUnit3.ConsoleRunner 3.15.2 before trying to use the nunit3-netcore-console.dll.

Option 1 was not an option because removing those references prevents me from running the tests in VS. Option 3 was not an option because I want to run on Linux without Mono.
